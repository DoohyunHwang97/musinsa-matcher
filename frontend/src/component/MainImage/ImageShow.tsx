import React, { useState } from "react";
import ItemModal from "./ItemModal";
const ImageShow = (props: any) => {
  const [modalOpen, setModalOpen] = useState(false);
  const showModal = () => {
    setModalOpen(true);
  };
  return (
    <div>
      <img
        data-testid="image"
        className="Mainimage"
        alt="img"
        src={props.src}
        onClick={showModal}
      ></img>
      {modalOpen && (
        <ItemModal
          id={props.id}
          src={props.src}
          setModalOpen={setModalOpen}
          URL={props.URL}
          name={props.name}
        />
      )}
      <h2>{props.name}</h2>
    </div>
  );
};
export default ImageShow;
