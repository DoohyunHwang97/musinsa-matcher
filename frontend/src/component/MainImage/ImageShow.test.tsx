import React from "react";
import renderer from "react-test-renderer";
import { NavLink } from "react-router-dom";
import { expect } from "@jest/globals";
import { mount } from "enzyme";
import ImageShow from "./ImageShow";

describe("ImageShow", () => {
  let component: any = null;

  it("initial render", () => {
    component = renderer.create(<ImageShow />);
  });
  it("render correct", () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe("<Imageshow />", () => {
  it("onClick", () => {
    const wrapper = mount(<ImageShow />);
    expect(wrapper.state("modalOpen")).toEqual(false);
    wrapper.find("img").simulate("click");
    expect(wrapper.state("modalOpen")).toEqual(true);
  });
});
