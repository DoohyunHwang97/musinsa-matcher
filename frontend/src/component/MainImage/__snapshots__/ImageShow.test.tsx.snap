// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageShow render correct 1`] = `
<div>
  <img
    alt="img"
    onClick={[Function]}
  />
  <h2 />
</div>
`;
